src/functional/list.h

```diff
@@ -0,0 +1,31 @@
+#ifndef LIST_H
+#define LIST_H
+
+#include <stddef.h>
+#include <stdbool.h>
+
+typedef struct list_t list_t;
+
+struct list_t {
+	void* head;
+	void* tail;
```

Comment:
Don't assume that I didn't know what a linked list is.
Yes, a list with a node removed is still a list.
But in C, the list is not named that way, otherwise a reference like `my_list->head` could be confusing on what it does.
Syntax like `tail(tail(tail(my_list)))` is more hellish.
Please use C way of referencing it, that is: `my_list->next->next->next`